frontend/web/components/import-export/FeatureImport.tsx

```diff
@@ -0,0 +1,303 @@
+    ProjectStore.getEnvs()?.[0]?.api_key,
+  )
+  const [tags, setTags] = useState<(number | string)[]>([])
+  const [search, setSearch] = useState()
+  const [page, setPage] = useState(0)
+  const env = ProjectStore.getEnvironment(environment)
+  const [file, setFile] = useState<File | null>(null)
+  const [fileData, setFileData] = useState<FeatureImportItem[] | null>(null)
+  const [currentProjectflags, setCurrentProjectflags] =
+    useState<ProjectFlag[]>()
+  const [currentFeatureStates, setCurrentFeatureStates] =
+    useState<Record<number, FeatureState>>()
+
+  useEffect(() => {
+    const callback = () => {
+      setCurrentProjectflags(FeatureListStore.getProjectFlags())
+      setCurrentFeatureStates(FeatureListStore.getEnvironmentFlags())
+    }
+    FeatureListStore.on('change', callback)
+
+    return () => {
+      FeatureListStore.off('change', callback)
+    }
+  }, [])
+  useEffect(() => {
+    if (previewEnvironment) {
+      AppActions.getFeatures(
+        projectId,
+        previewEnvironment,
+        true,
+        null,
+        null,
+        page,
+        {
+          search,
+          tags: tags?.length ? tags : undefined,
+        },
+      )
+    }
+  }, [previewEnvironment, tags, search, projectId, page])
+
+  const [createImport, { error, isLoading }] =
+    useCreateFlagsmithProjectImportMutation({})
+
+  const onSubmit = () => {
+    if (env && file) {
+      createImport({
+        environment_id: env.id,
+        file,
+        strategy,
+      }).then((res) => {
+        if (res?.data) {
+          toast('Your import is processing')
+        } else {
+          toast('Failed to import flags')
+        }
+      })
+    }
+  }
+
+  const [strategy, setStrategy] = useState<ImportStrategy>('SKIP')
+
+  useEffect(() => {
+    AppActions.getFeatures(projectId, environment, true, null, null, page, {
+      search,
+      tags: tags?.length ? tags : undefined,
+    })
+  }, [projectId, tags])
+  const {
+    featureStates,
+    projectFlags,
+  }: {
+    projectFlags: ProjectFlag[] | null
+    featureStates: FeatureState[] | null
+  } = useMemo(() => {
+    if (fileData) {
+      const createdDate = new Date().toISOString()
+      const existingFlags: ProjectFlag[] =
+        !!fileData &&
+        !!currentFeatureStates &&
+        currentProjectflags?.map((projectFlag, i) => {
+          const featureState = currentFeatureStates[projectFlag.id]
+          const importItem = fileData.find((v) => v.name === projectFlag.name)
+          let newValue = featureState.value
+          let newEnabled = featureState.enabled
+          if (strategy === 'OVERWRITE_DESTRUCTIVE' && importItem) {
+            if (environment === previewEnvironment) {
+              newEnabled = importItem.enabled
+              newValue = importItem.value
+            } else {
+              newEnabled = importItem.default_enabled
+              newValue = importItem.initial_value
+            }
+          }
```

Comment:
This block needs a comment. Two or three times I thought you implemented it incorrectly, but it's done right. But it will be hard for others to follow the logic in the future.